fish-rust/src/trace.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    common::{escape_string, EscapeFlags},
+    ffi::{self, parser_t, wcharz_t},
+    wchar::{self, L},
+    wchar_ffi::{wstr, WCharToFFI},
+};
+use std::sync::atomic;
+
+#[cxx::bridge]
+mod ffi2 {
+    extern "C++" {
+        include!("wutil.h");
+        include!("parser.h");
+        type wcharz_t = super::wcharz_t;
+        type parser_t = super::parser_t;
+    }
+
+    extern "Rust" {
+        fn trace_set_enabled(do_enable: bool);
+        fn trace_enabled(parser: &parser_t) -> bool;
+        fn rust_trace_argv(parser: &parser_t, command: wcharz_t, args: &Vec<wcharz_t>);
+        fn trace_if_enabled(parser: &parser_t, command: wcharz_t);
```

Comment:
Let's name this `rust_trace_if_enabled`. The convention is that Rust functions called from C++ should start with `rust_`

Reply:
All FFI functions are now prefixed with `rust_`